sdks/java/extensions/sbe/src/main/java/org/apache/beam/sdk/extensions/sbe/SbeLogicalTypes.java

```diff
@@ -0,0 +1,264 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sbe;
+
+import static org.apache.beam.sdk.extensions.sbe.Schemas.TZ_TIME_SCHEMA;
+import static org.apache.beam.sdk.extensions.sbe.Schemas.UTC_TIME_SCHEMA;
+
+import java.time.LocalDate;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+import org.apache.beam.sdk.extensions.sbe.TimeValues.TZTimeOnlyValue;
+import org.apache.beam.sdk.extensions.sbe.TimeValues.TZTimestampValue;
+import org.apache.beam.sdk.extensions.sbe.TimeValues.UTCTimeOnlyValue;
+import org.apache.beam.sdk.extensions.sbe.TimeValues.UTCTimestampValue;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import org.apache.beam.sdk.schemas.Schema.LogicalType;
+import org.apache.beam.sdk.schemas.logicaltypes.PassThroughLogicalType;
+import org.apache.beam.sdk.values.Row;
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.checker.nullness.qual.UnknownKeyFor;
+
+/**
+ * Classes that represent various SBE semantic types.
+ *
+ * <p>Not all of SBE's semantic types are represented here, as some can be reasonably represented in
+ * Beam schemas with just a primitive.
+ */
+@Experimental(Kind.SCHEMAS)
+public final class SbeLogicalTypes {
+  public static final Uint8 UINT_8 = new Uint8();
+  public static final Uint16 UINT_16 = new Uint16();
+  public static final Uint32 UINT_32 = new Uint32();
+  public static final Uint64 UINT_64 = new Uint64();
+
+  public static final UTCTimestamp UTC_TIMESTAMP = new UTCTimestamp();
+  public static final UTCTimeOnly UTC_TIME_ONLY = new UTCTimeOnly();
+  public static final UTCDateOnly UTC_DATE_ONLY = new UTCDateOnly();
+  public static final TZTimestamp TZ_TIMESTAMP = new TZTimestamp();
+  public static final TZTimeOnly TZ_TIME_ONLY = new TZTimeOnly();
+  public static final LocalMktDate LOCAL_MKT_DATE = new LocalMktDate();
+
+  // Default argument type values
+  private static final String DEFAULT_STRING_ARG = "";
+
+  private SbeLogicalTypes() {}
+
+  // Unsigned types are all stored at the next highest value. This prevents unexpected behavior
+  // when reading and likely has negligible space impact.
```

Comment:
My main concern with doing the protobuf approach is that conversion from Row to JSON will use the base value, which will be signed. Since conversion to JSON is probably going to be a standard part of any pipeline using this, I'm a bit concerned that it'll lead to a lot of unexpected output.

Or am I misunderstanding how `ToJson` works?